src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -318,20 +319,21 @@ citus_table_size(PG_FUNCTION_ARGS)
 	Oid relationId = PG_GETARG_OID(0);
 	bool failOnError = true;
 	SizeQueryType sizeQueryType = TABLE_SIZE;
-	uint64 tableSize = 0;
+	uint64 relationSize = 0;
 
-	if (!DistributedTableSize(relationId, sizeQueryType, failOnError, &tableSize))
+	/* We do not check if relation is really a table, like PostgreSQL is doing. */
+	if (!DistributedRelationSize(relationId, sizeQueryType, failOnError, &relationSize))
 	{
 		Assert(!failOnError);
 		PG_RETURN_NULL();
 	}
 
-	PG_RETURN_INT64(tableSize);
+	PG_RETURN_INT64(relationSize);
 }
 
 
 /*
- * citus_relation_size accept a table name and returns a relation's 'main'
+ * citus_relation_size accept a relation name and returns a relation's 'main'
```

Comment:
```suggestion
 * citus_relation_size accept a distributed relation and returns its 'main'
```

Reply:
I am unsure about Citus words: if the relation is an index, is it a distributed relation ? or only the table is flagged distributed ?

Reply:
Well, normally we don't call indexes on distributed tables as "distributed tables" but it might make sense to do that in this context.

Reply:
added